### DISCUSSION TITLE
Can sahi do slice predictions without torch dependencies?

### DISCUSSION THREAD

**Discussion by zhaoruibing** [obss/sahi#1006] [General]
I am trying to deploy a YOLOv8 model with `get_sliced_prediction()`, pack everything into a small container.
`get_sliced_prediction()` calls postprocess functions, and these postprocess functions in `sahi.postprocess.combine` depends on torch.

sahi is said to be 'lightweight' and in [this discussion](https://github.com/obss/sahi/discussions/659) independent of torch.
> We wanted sahi to work independent from torch thats why all detectors accept numpy or str as image input 👍

The question is, could sahi do slice predictions without torch dependencies?
Or this is something sahi plans to do in the future?

**Comment by HowcanoeWang** [comment]
I have the same requirements, so I manually modify the source code to change all torch based to numpy based (just for yolo detection, I also removed several unnecessary codes), you can check the source code here (based on v0.11.20): 

https://github.com/UTokyo-FieldPhenomics-Lab/EasyAMS/tree/0e74099e311052943b280eb402f61caa6ad752d9/easyams/sahi_onnx

The modification can be checked at this commit:

https://github.com/UTokyo-FieldPhenomics-Lab/EasyAMS/commit/0e74099e311052943b280eb402f61caa6ad752d9

**Comment by onuralpszr** [comment]
Next release will be not gonna include torch and all of the process can be done via "numpy" but it can be use via torch or numba as well. So I make sure it is now optional as well. But please try and let us know, thank you.

**Comment by onuralpszr** [comment]
SAHI 0.12.0 is now works without torch dependencies by default.
